File: lab6/matrix.c
/*********************************************/
/*                                           */
/*                                           */
/*          Lavrinenko Nikita IO-62          */
/*                                           */
/*                                           */
/*********************************************/

#include "opp.h"

float   **Fill_Matrix(float num)
{
    float   **matr;

    if (!(matr = (float **)malloc(sizeof(float *) * (N + 1))))
        return (NULL);
    for (int i = 0; i < N; i++)
        matr[i] = Fill_Vector(num);
    matr[N] = NULL;
    return (matr);
}

float   **Add_Matr(float **Matr1, float **Matr2)
{
    for (int i = 0; i < N; i++)
        Matr1[i] = Add_Vec(Matr1[i], Matr2[i]);
    return (Matr1);
}

float **Mull_Matr(float **Matr1, float **Matr2)
{
    float **matr;

    matr = Fill_Matrix(0.0);
    for (int i = 0; i < N; i ++)
        for (int j = 0; j < N; j++)
            for (int k = 0; k < N; k++)
                matr[i][j] += Matr1[i][k] * Matr2[k][j];
    return (matr);
}

float   **Mull_MatrxNum(float a, float **MB)
{
    float **matr;

    matr = Fill_Matrix(0.0);
    for (int i = 0; i < N; i++)
        for (int j = 0; j < N; j++)
            matr[i][j] = a * MB[i][j];
    return (matr);
}

float   **Tranc_Matr(float **Matr)
{
    float **ResMatr;

    ResMatr = Fill_Matrix(0.0);
    for (int i = 0; i < N; i++)
        for (int j = 0; j < N; j++)
            ResMatr[i][j] = Matr[j][i];
    return (Matr);
}

void    Print_Matr(float ** Matr)
{
    for (int i = 0; i < N; i++)
        Print_Vec(Matr[i]);
    printf("\n");
}

File: lab2/Matrix.java
import java.util.Arrays;

class Matrix {

    // ?????? ???????
    private static Integer Size = Lab2.N;

    // ?????????? ???????, ?????????? ?????????? ???????
    public double[][] Random_Matrix() {
        double[][] Result = new double[Size][Size];
        for (int i = 0; i < Size; i++) {
            for (int j = 0; j < Size; j++) {
                Result[i][j] = Math.random() * 20.0;
            }
        }
        return Result;
    }

    // ?????????? ???????, ???????? ??????
    public double[][] Fill_Matrix(double number) {

        double[][] Result = new double[Size][Size];

        for (int i = 0; i < Size; i++) {
            for (int j = 0; j < Size; j++) {
                Result[i][j] = number;
            }
        }
        return Result;
    }

    // ??????????? ???????
    public double[][] Add_Matr(double[][] Matr1, double[][] Matr2) {
        for (int i = 0; i < Size; i++) {
            for (int j = 0; j < Size; j++) {
                Matr1[i][j] = Matr1[i][j] + Matr2[i][j];
            }
        }
        return Matr1;
    }

    // ????????? ??????
    public double[][] Mull_Matr(double[][] MA, double[][] MB) {

        double[][] ResultMatrix = new double[Size][Size];

        for (int i = 0; i < Size; i++) {
            for (int j = 0; j < Size; j++) {
                for (int k = 0; k < Size; k++) {

                    ResultMatrix[i][j] += MA[i][k] * MB[k][j];

                }
            }
        }
        return ResultMatrix;
    }

    // ????????? ??????? ?? ?????
    public double[][] Mull_MatrxNum(double a, double[][] MB) {

        double[][] ResultMatrix = new double[Size][Size];

        for (int i = 0; i < Size; i++) {
            for (int j = 0; j < Size; j++) {

                ResultMatrix[i][j] = a * MB[i][j];

            }
        }
        return ResultMatrix;
    }

    // ???????????????? ???????
    public double[][] Tranc_Matr(double[][] Matr){

        double[][] ResMatr = new double[Size][Size];

        for (int i = 0; i < Size; i++){
            for (int j = 0; j < Size; j++){

                ResMatr[i][j] = Matr[j][i];

            }        
        }
        return ResMatr;
    }

    // ????? ???????
    public void Print_Matrix(double[][] matrix) {
        System.out.println("Result:");
        for (int i = 0; i < matrix.length; i++) {

            System.out.println(Arrays.toString(matrix[i]));

        }
    }
}